test/puppetlabs/trapperkeeper/optional_deps_test.clj

```diff
@@ -32,6 +32,83 @@
                                             couplet))))
 
 (deftest optional-deps-test
+  (testing "when using defservice"
+    (testing "with a protocol"
+      (defservice poetry-svc-w-protocol
+        PoetryService
+        {:required [HaikuService]
+         :optional [SonnetService]}
+        (init [this ctx] ctx)
+        (get-haiku [this] "haiku")
+        (get-sonnet [this] "sonnet"))
```

Comment:
I also hate this. But the point of these tests is to make sure that defservice compiles ok. I figured having them fail during macro compilation near an actual (testing) line would make the horrific task of debugging test failures here slightly less bad.